python-frontend/src/main/java/org/sonar/python/semantic/v2/SymbolsModuleTypeProvider.java

```diff
@@ -88,147 +68,20 @@ private static String getModuleFqnString(List<String> moduleFqn) {
   }
 
   private Optional<ModuleType> createModuleTypeFromProjectLevelSymbolTable(String moduleName, String moduleFqn, ModuleType parent) {
-    return Optional.ofNullable(projectLevelSymbolTable.getSymbolsFromModule(moduleFqn))
-      .map(projectModuleSymbols -> createModuleFromSymbols(moduleName, parent, projectModuleSymbols));
+    var retrieved = projectLevelSymbolTable.getDescriptorsFromModule(moduleFqn);
+    if (retrieved == null) {
```

Comment:
can't it be converted just to `Optional.ofNullable(...)...`, then you simply don't need this condition like:
```
return Optional.ofNullable(projectLevelSymbolTable.getDescriptorsFromModule(moduleFqn))
 .map(retrieved -> retrieved.stream().collect(Collectors.toMap(Descriptor::name, d -> anyDescriptorToPythonTypeConverter.convert(d, TypeOrigin.LOCAL))))
 .map(members -> new ModuleType(moduleName, parent, members))
```